### PR TITLE
Feature/doc updates

### PR DESCRIPTION
**Pull request by rddesmond**
Update the validation and documentation consistency.  Remove 'landcover' from the short-form documentation.

### CODE REVIEW

File: doc/REFERENCE.md [hunk 1]
```diff
@@ -219,84 +215,161 @@ AWS_ACCESS_KEY_ID=[AWS_ACCESS_KEY_ID] AWS_SECRET_ACCESS_KEY=[AWS_SECRET_ACCESS_K
...
+
+By default, no regions are excluded.  If any of these options are given, a filter is built.  If the
+first action is "exclude", the filter is initialized to include the bounding box.
+
+`--exclude-region` will subtract geometries contained within the supplied path(s) to the from the search region.
+
+`--include-region` will add the geometries contained within the supplied path(s) from the search region.
+
+`--region` can be used to chain together multiple includes and excludes.  This is important because the order of
+operations matters and each of the specific versions (i.e. `--exclude-region` and `-include-region`) may only be
```
Comment (avitebskiy) [inline doc/REFERENCE.md]: `--include-region` and `--exclude-region` may be specified more than once.

File: doc/REFERENCE.md [hunk 2]
```diff
@@ -219,84 +215,161 @@ AWS_ACCESS_KEY_ID=[AWS_ACCESS_KEY_ID] AWS_SECRET_ACCESS_KEY=[AWS_SECRET_ACCESS_K
...
-## Logging Options
+
+### Logging Options
 
 ##### --log
 
-This option specifies a log file that _OpenSpaceNet_ writes to. Optionally a log level can be specified. Permitted log level values are
-`trace`, `debug`, `info`, `warning`, `error`, and `fatal`. The default log level is `info`.
+This option specifies a log file that _OpenSpaceNet_ writes to. Optionally, a log level can be specified. Permitted
+log level values are `trace`, `debug`, `info`, `warning`, `error`, and `fatal`. The default log level is `info`.
```
Comment (avitebskiy) [inline doc/REFERENCE.md]: Default log level for file output is `debug`.

File: doc/REFERENCE.md [hunk 3]
```diff
@@ -373,14 +449,15 @@ Running _OpenSpaceNet_ with this file
 
 is the same as running _OpenSpaceNet_ with these options:
 ```
-./OpenSpaceNet detect --service dgcs --token=abcd-efgh-ijkl-mnop-qrst-uvxyz --credentials username:password --bbox=-84.44579 33.63404 -84.40601 33.64853 --model airliner.gbdxm --output atl_detected.shp --confidence=99 --step-size=15 --num-downloads=200 --nms
+./OpenSpaceNet detect --service dgcs --token=abcd-efgh-ijkl-mnop-qrst-uvxyz --credentials username:password --bbox=-84.44579 33.63404 -84.40601 33.64853 --model airliner.gbdxm --output atl_detected.shp --confidence=99 --window-step=15 --num-downloads=200 --nms
```
Comment (avitebskiy) [inline doc/REFERENCE.md]: Should be
```
./OpenSpaceNet detect --service dgcs --token abcd-efgh-ijkl-mnop-qrst-uvxyz --credentials username:password --bbox -84.44579 33.63404 -84.40601 33.64853 --model airliner.gbdxm --output atl_detected.shp --confidence 99 --window-step 15 --num-downloads 200 --nms
```

File: src/OpenSpaceNetArgs.cpp [hunk 4]
```diff
@@ -439,183 +438,166 @@ void OpenSpaceNetArgs::printUsage(Action action) const
     cout << endl;
 }
 
+enum ArgUse {
```
Comment (avitebskiy) [inline src/OpenSpaceNetArgs.cpp]: Interesting idea, it may be worth to make it more integral with `boost::program_options` in the future.

Comment (rddesmond) [inline src/OpenSpaceNetArgs.cpp]: The problem with that is that these are all dependent on other arguments.  E.g. setting `--image` means `--map-id` is ignored.  That isn't known a-priori or causally.

Comment (avitebskiy) [inline src/OpenSpaceNetArgs.cpp]: I know. We'd need a mechanism to set those easily as well.

Comment (avitebskiy) [inline src/OpenSpaceNetArgs.cpp]: I know. We'd need a mechanism to set those easily as well.

**Pull request merged**